### PR TITLE
ISPN-3795 QueryInterceptor incorrectly relies on the return value of a RemoveCommand

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-3795

### CODE REVIEW

**Comment by karesti** [comment]
looks good for me if CI is ok too

**Comment by gustavonalle** [comment]
There are some test failures, looking...

**Comment by gustavonalle** [comment]
Should be fine now

File: integrationtests/elasticsearch-indexmanager-it/src/test/java/org/infinispan/query/it/ClusteredCacheWithElasticsearchIndexManagerIT.java [hunk 1]
```diff
@@ -1,15 +1,21 @@
 package org.infinispan.query.it;
 
+import org.apache.lucene.search.MatchAllDocsQuery;
```
Comment (wburns) [inline integrationtests/elasticsearch-indexmanager-it/src/test/java/org/infinispan/query/it/ClusteredCacheWithElasticsearchIndexManagerIT.java]: Unused imports

File: query/src/test/java/org/infinispan/query/test/MemberType.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.infinispan.query.test;
+
+/**
+ * @since 9.1
+ */
+public enum MemberType {
```
Comment (rvansa) [inline query/src/test/java/org/infinispan/query/test/MemberType.java]: Use `o.i.distribution.Ownership`

Comment (gustavonalle) [inline query/src/test/java/org/infinispan/query/test/MemberType.java]: definitely 👍

Comment (gustavonalle) [inline query/src/test/java/org/infinispan/query/test/MemberType.java]: I will handle the tests with flag in a follow-up JIRA  https://issues.jboss.org/browse/ISPN-7828

**Comment by gustavonalle** [comment]
@rvansa I will handle the tests with flag in a follow-up JIRA https://issues.jboss.org/browse/ISPN-7828

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java [hunk 3]
```diff
@@ -235,6 +249,115 @@ public void testRemoved() throws Exception {
...
+
+      Predicate<Cache<?, ?>> predicate = null;
+      switch (ownership) {
+         case PRIMARY:
+            predicate = c -> c.getAdvancedCache().getRpcManager().getAddress().equals(distribution.primary());
+            break;
+         case BACKUP:
+            predicate = c -> {
+               Address address = c.getAdvancedCache().getRpcManager().getAddress();
+               return distribution.writeOwners().contains(address) && !distribution.primary().equals(address);
```
Comment (rvansa) [inline query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java]: use `distribution.writeBackups()`

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java [hunk 4]
```diff
@@ -235,6 +249,115 @@ public void testRemoved() throws Exception {
...
+            };
+            break;
+         case NON_OWNER:
+            predicate = c -> {
+               Address address = c.getAdvancedCache().getRpcManager().getAddress();
+               return !distribution.writeOwners().contains(address) && !distribution.primary().equals(address);
```
Comment (rvansa) [inline query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java]: primary is always amongst the writeOwners, no need to test separately

**Comment by gustavonalle** [comment]
@rvansa change in the value type is already tested. Regarding passivation, I'm aware it will fail, see [comments](https://issues.jboss.org/browse/ISPN-7806)

**Comment by gustavonalle** [comment]
Failures not related

**Comment by anistor** [comment]
LGTM too.

**Comment by anistor** [comment]
Integrated in master. Thanks @gustavonalle !

**Comment by anistor** [comment]
@gustavonalle Do we want this in 9.0.x too ?

**Comment by gustavonalle** [comment]
Yes, please!

**Pull request closed without merging**